src/conn.ts

```diff
@@ -167,8 +159,14 @@ export class Conn {
         } else {
           data = funcReturn;
         }
-        isCanceled = data === false;
-        if (data !== undefined && data !== false) {
+        if (!isCanceled) { // Wait for the first occurrence 
+          isCanceled = data === false;
+        } else {
+          if (data === true) {
+            isCanceled = false; // Allow following middlewares to un cancel packet that have been canceled already
+          }
+        }
```

Comment:
```suggestion
        isCanceled = isCanceled ? data !== true : data === false;
```

Reply:
This is not the same logic tho. `data !== true` is not the same as `data === true`. And the value of isCanceled should not change when isCanceled is false and data is not equal to true.